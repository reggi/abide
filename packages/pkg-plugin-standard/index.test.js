import pluginStandard, {ifLengthAddProp} from './index'

test('ifLengthAddProp', () => {
  expect(ifLengthAddProp('prop', [])).toEqual({})
  expect(ifLengthAddProp('prop', {})).toEqual({})
  expect(ifLengthAddProp('prop', ['hello'])).toEqual({'prop': ['hello']})
  expect(ifLengthAddProp('prop', {'hello': 'world'})).toEqual({'prop': {'hello': 'world'}})
})

test('pluginStandard: default', () => {
  const results = pluginStandard()
  expect(results).toMatchSnapshot()
})

test('pluginStandard: babel', () => {
  const args = {opt: {babel: true}}
  const results = pluginStandard(args)
  expect(results).toMatchSnapshot()
})

test('pluginStandard: jest', () => {
  const args = {opt: {jest: true}}
  const results = pluginStandard(args)
  expect(results).toMatchSnapshot()
})

test('pluginStandard: default with pkg', () => {
  const args = {
    pkg: {
      existing: true,
      scripts: {
        existing: true
      },
      devDependencies: {
        existing: true
      }
    }
  }
  const results = pluginStandard(args)
  expect(results).toMatchSnapshot()
})
