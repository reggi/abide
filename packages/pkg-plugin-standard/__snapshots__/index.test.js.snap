// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pluginStandard: babel 1`] = `
Object {
  "devDependencies": Object {
    "babel-eslint": "^8.2.2",
    "standard": "^11.0.0",
  },
  "scripts": Object {
    "standard": "standard",
  },
  "standard": Object {
    "parser": "babel-eslint",
  },
}
`;

exports[`pluginStandard: default 1`] = `
Object {
  "devDependencies": Object {
    "standard": "^11.0.0",
  },
  "scripts": Object {
    "standard": "standard",
  },
}
`;

exports[`pluginStandard: default with pkg 1`] = `
Object {
  "devDependencies": Object {
    "existing": true,
    "standard": "^11.0.0",
  },
  "existing": true,
  "scripts": Object {
    "existing": true,
    "standard": "standard",
  },
}
`;

exports[`pluginStandard: jest 1`] = `
Object {
  "devDependencies": Object {
    "standard": "^11.0.0",
  },
  "scripts": Object {
    "standard": "standard",
  },
  "standard": Object {
    "globals": Array [
      "beforeAll",
      "afterAll",
      "expect",
      "test",
      "beforeEach",
      "afterEach",
      "it",
      "jest",
    ],
  },
}
`;
